txcache/crossTxCache.go

```diff
@@ -93,7 +98,11 @@ func (cache *CrossTxCache) Peek(key []byte) (value interface{}, ok bool) {
 
 // RemoveTxByHash removes tx by hash
 func (cache *CrossTxCache) RemoveTxByHash(txHash []byte) bool {
-	return cache.RemoveWithResult(txHash)
+	ok := cache.RemoveWithResult(txHash)
+	if ok {
+		go cache.notifyEvictionHandlers(txHash)
```

Comment:
if we remove a lot of transactions, a high number of goroutines will be created. 
`cache.notifyEvictionHandlers(txHash)` could have been called without the `go` prefix and it can contain a queue-based worker pool that limits the number of goroutines that are spawned.

Reply:
good suggestion, implemented it

Reply:
removed the implementation and used library as suggested